File: B-YEP-400-LIL-4-1-zappy-azan.jean-baptiste-main/B-YEP-400-LIL-4-1-zappy-azan.jean-baptiste-main/ai/src/mouv.cpp
#include "mouv.hpp"
#include "IA.hpp"
#include <iostream>
#include <string>
#include "MyIA.hpp"

void handle_movement(int s, IA& ia) {
    if (ia.nb_command_send < 10 && !ia.steps.empty()) {
        std::string command = ia.steps[0] + "\n";
        send(s, command.c_str(), command.length(), 0);
        ia.nb_command_send++;
        std::cout << "commande exec : " << command;
        ia.steps.erase(ia.steps.begin());
        handle_movement(s, ia);
    }
}

File: B-YEP-400-LIL-4-1-zappy-azan.jean-baptiste-main/B-YEP-400-LIL-4-1-zappy-azan.jean-baptiste-main/IA/mouv.py
def handle_movement(s, ia):
    if (ia.nb_command_send <10 and ia.steps):
        command = ia.steps[0].encode() + b'\n'
        s.sendall(command)
        ia.nb_command_send += 1
        print("commande exec : ", command)
        ia.steps.pop(0)
        handle_movement(s, ia)